ZydisEncoder/generate_encoder_types.py

```diff
@@ -0,0 +1,255 @@
+        'stddef.h': '',
+        'errno.h': '#define errno 0',
+        'stdarg.h': """
+            typedef char* va_list;
+            #define va_start(...)
+            #define va_arg(...)
+            #define va_end(...)
+            #define va_copy(...)""",
+        'stdio.h': """
+            typedef struct _FILE
+            {
+                void* Dummy;
+            } FILE;
+
+            #define stdin ((FILE *)0)
+            #define stdout ((FILE *)0)
+            #define stderr ((FILE *)0)
+            #define fputc(...)
+            #define fprintf(...)
+            #define printf(...)
+            #define putc(...)
+            #define puts(...)
+            #define scanf(...)
+            #define sscanf(...)
+            #define vsnprintf(...)
+        """,
+        'stdlib.h': """
+            #define calloc(...)
+            #define free(...)
+            #define malloc(...)
+            #define realloc(...)
+        """,
+        'string.h': """
+            #define memchr(...)
+            #define memcmp(...)
+            #define memcpy(...)
+            #define memmove(...)
+            #define memset(...)
+            #define strcat(...)
+            #define strchr(...)
+            #define strcmp(...)
+            #define strcoll(...)
+            #define strcpy(...)
+            #define strcspn(...)
+            #define strlen(...)
+            #define strncat(...)
+            #define strncmp(...)
+            #define strncpy(...)
+            #define strpbrk(...)
+            #define strrchr(...)
+            #define strspn(...)
+            #define strstr(...)
+            #define strtok(...)
+            #define strxfrm(...)
+        """,
+        'stdint.h': """
+            typedef signed char        int8_t;
+            typedef short              int16_t;
+            typedef int                int32_t;
+            typedef long long          int64_t;
+            typedef unsigned char      uint8_t;
+            typedef unsigned short     uint16_t;
+            typedef unsigned int       uint32_t;
+            typedef unsigned long long uint64_t;
+            typedef unsigned int       size_t;
+            typedef int                ptrdiff_t;
+            typedef int                intptr_t;
+            typedef unsigned int       uintptr_t;
+
+            #define INT8_MIN           (-127 - 1)
+            #define INT16_MIN          (-32767 - 1)
+            #define INT32_MIN          (-2147483647 - 1)
+            #define INT64_MIN          (-9223372036854775807 - 1)
+            #define INT8_MAX           127
+            #define INT16_MAX          32767
+            #define INT32_MAX          2147483647
+            #define INT64_MAX          9223372036854775807
+            #define UINT8_MAX          0xff
+            #define UINT16_MAX         0xffff
+            #define UINT32_MAX         0xffffffff
+            #define UINT64_MAX         0xffffffffffffffff""",
+    }
```

Comment:
I think you shouldn't even need these if you just define `ZYAN_NO_LIBC`, making Zycore fallback to our own types.

Reply:
There are different paths to take (I didn't explore them really) but all of them have disadvantages. `ZYAN_NO_LIBC` does fallback to compiler-specific stuff, so I would need to do some faking for that. Honestly I've chosen libc path because initially I was parsing `Encoder.h` which needs less fakes but then I realized that I need `ZYDIS_ENCODABLE_PREFIXES` and that is inside 'Encoder.c'. This makes libc path much uglier but I sticked with it anyway.

Reply:
I tried to evaluate that out of curiosity. `MSVC` path is impossible due to custom integer prefixes. `GCC` path looks better but here comes the problem: `typedef va_list ZyanVAList;`. This won't parse in `ZYAN_NO_LIBC` mode because `va_list` isn't defined. I can fix that by defining it here but isn't that a bug for Zydis?